### PR TITLE
[ANDROAPP-5409] update SDK to 1.9.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/data/dhislogic/DhisEventUtils.kt [hunk 1]
```diff
@@ -8,13 +8,13 @@ class DhisEventUtils @Inject constructor(val d2: D2) {
...
+        val stage = d2.programModule().programStages().uid(event?.programStage())
             .blockingGet()
-        val program = d2.programModule().programs().uid(stage.program()?.uid())
+        val program = d2.programModule().programs().uid(stage?.program()?.uid())
             .blockingGet()
-        val hasCoordinates = stage.featureType() != null && stage.featureType() != FeatureType.NONE
+        val hasCoordinates = stage?.featureType() != null && stage.featureType() != FeatureType.NONE
         val hasNonDefaultCatCombo = d2.categoryModule().categoryCombos()
-            .uid(program.categoryComboUid()).blockingGet().isDefault != true
+            .uid(program?.categoryComboUid()).blockingGet()?.isDefault != true
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/dhislogic/DhisEventUtils.kt:17]: is there a possibility to call `d2.categoryModule().categoryCombos()uid(**null**).blockingGet` ? . Are null parameters allowed?

File: app/src/main/java/org/dhis2/data/forms/dataentry/ValueStoreImpl.kt [hunk 2]
```diff
@@ -131,8 +131,8 @@ class ValueStoreImpl(
                 EntryMode.DE -> {
                     val event = d2.eventModule().events().uid(recordUid).blockingGet()
                     val enrollment = d2.enrollmentModule().enrollments()
-                        .uid(event.enrollment()).blockingGet()
-                    enrollment.trackedEntityInstance()
+                        .uid(event?.enrollment()).blockingGet()
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/forms/dataentry/ValueStoreImpl.kt:134]: shouldn't launch an exception uid(NULL)?. maybe is better `event?.let { d2.enrollmentModule().enrollments().uid(it) }`

File: app/src/main/java/org/dhis2/data/forms/dataentry/ValueStoreImpl.kt [hunk 3]
```diff
@@ -291,11 +291,11 @@ class ValueStoreImpl(
 
     private fun deleteDataElementValue(field: String, optionUid: String): StoreResult {
         val option = d2.optionModule().options().uid(optionUid).blockingGet()
-        val possibleValues = arrayListOf(option.name()!!, option.code()!!)
+        val possibleValues = arrayListOf(option?.name(), option?.code()).filterNotNull()
         val valueRepository =
             d2.trackedEntityModule().trackedEntityDataValues().value(recordUid, field)
         return if (valueRepository.blockingExists() &&
-            possibleValues.contains(valueRepository.blockingGet().value())
+            possibleValues.contains(valueRepository.blockingGet()?.value())
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/forms/dataentry/ValueStoreImpl.kt:298]: I think that this methods always returns false is option is null, so maybe you can return false after the line 293 if it is null.

File: app/src/main/java/org/dhis2/data/forms/dataentry/ValueStoreImpl.kt [hunk 4]
```diff
@@ -305,11 +305,11 @@ class ValueStoreImpl(
 
     private fun deleteAttributeValue(field: String, optionUid: String): StoreResult {
         val option = d2.optionModule().options().uid(optionUid).blockingGet()
-        val possibleValues = arrayListOf(option.name()!!, option.code()!!)
+        val possibleValues = arrayListOf(option?.name(), option?.code()).filterNotNull()
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/forms/dataentry/ValueStoreImpl.kt:308]: I think that this methods always returns false is option is null, so maybe you can return false after the line 307 if it is null.

File: app/src/main/java/org/dhis2/data/service/SyncPresenterImpl.kt [hunk 5]
```diff
@@ -402,13 +402,13 @@ class SyncPresenterImpl(
         val enrollment = d2.enrollmentModule().enrollments().uid(uid).blockingGet()
         Completable.fromObservable(
             d2.trackedEntityModule().trackedEntityInstances()
-                .byUid().eq(enrollment.trackedEntityInstance()!!)
-                .byProgramUids(listOf(enrollment.program()!!))
+                .byUid().eq(enrollment?.trackedEntityInstance())
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/data/service/SyncPresenterImpl.kt:405]: can this lead to an exception? `byUid().eq(enrollment?.trackedEntityInstance())` if enrollment is null?

**Review by mmmateos — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Pull request merged**